### PR TITLE
 Get-DbaSsisExecutionHistory Goodies #3832

### LINKED ISSUE #3832: Get-DbaSsisExecutionHistory, various goodies
### Summary of new feature
- It sorely misses a -Since to filter out old executions
- It should not use string interpolation when building the query
- It has no tests

### Proposed technical details (if applicable)

Use parameters to avoid SQL Injections

### Latest version of dbatools as of writing

0.9.376

### PR DESCRIPTION
**Pull request by ChrisTuckerNM**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #3832)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
 
Goodies for Get-DbaSsisExecutionHistory
### Commands to test

Examples

### CODE REVIEW

File: functions/Get-DbaSsisExecutionHistory.ps1 [hunk 1]
```diff
@@ -85,38 +92,65 @@ function Get-DbaSsisExecutionHistory {
...
         else {
             $statusq = ''
         }
-        
+
+        #construct parameterized collection predicate for project array
         if ($Project) {
-            $csv = "`"" + ($Project -join '","') + "`""
-            $projectq = "AND e.[ProjectName] in ($csv)"
+            $projectq = "`n`t`tAND ( 1=0 "
```
Comment (wsmelton) [inline functions/Get-DbaSsisExecutionHistory.ps1]: We are not all that concerned with formatting in T-SQL....so no real reason to include all the tabs. The new line is added based on building the variable.

So if you have:
```powershell
"$var1
$var2
$var3
"
```
They will all be on a new line anyway.

Just an FYI....

**Comment by niphlod** [comment]
looks fine, even if I'd have gone for IN instead of OR but .... just personal preference. 
Ideas for a sane default value for -Since (i.e.  even without -Since passed, filter 1 week worth ) ?

File: functions/Get-DbaSsisExecutionHistory.ps1 (hunk 1 above)
Comment (ChrisTuckerNM) [reply functions/Get-DbaSsisExecutionHistory.ps1]: Pedantic's, ok if I keep it that way?

**Comment by ChrisTuckerNM** [comment]
Not sure how to do this with in and multiple choices to be honest, I felt like this was cleaner.

File: functions/Get-DbaSsisExecutionHistory.ps1 (hunk 1 above)
Comment (ChrisTuckerNM) [reply functions/Get-DbaSsisExecutionHistory.ps1]: And all that ugly white-space for parameters that have no value  ewwww.

**Comment by potatoqualitee** [comment]
thanks much, @ChrisTuckerNM !

**Comment by potatoqualitee** [comment]
(you guys can continue to talk, even tho i merged 👍)

File: functions/Get-DbaSsisExecutionHistory.ps1 (hunk 1 above)
Comment (wsmelton) [reply functions/Get-DbaSsisExecutionHistory.ps1]: It will stay until someone cleans it up ;)

**Pull request merged**